nibabel/bv_vtc.py

```diff
@@ -0,0 +1,250 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+# ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+# ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Reading / writing functions for Brainvoyager (BV) VTC files.
+
+for documentation on the file format see:
+http://support.brainvoyager.com/installation-introduction/23-file-formats/379-users-guide-23-the-format-of-vtc-files.html
+
+Author: Thomas Emmerling
+"""
+
+import numpy as np
+from .bv import BvError, BvFileHeader, BvFileImage
+from .spatialimages import HeaderDataError
+from .batteryrunners import Report
+from .externals import OrderedDict
+
+
+def _make_vtc_hdrDict():
+    """Helper for creating a VTC header OrderedDict with default parameters.
+
+    Create an OrderedDict that contains keys with the header fields.
+    The value of each field is a dict containing the DataType (dt), a default
+    value (default) and the actual value (value).
+
+    b := signed char (1 byte)
+    h := signed short integer (2 bytes)
+    s := string (variable bytes)
+    f := float (4 bytes)
+    """
+    VTC_HEADER_PROTO = OrderedDict([
+        ('version', {'dt': '<h', 'default': 3}),
+        ('fmr', {'dt': '<s', 'default': ''}),
+        ('nPrt', {'dt': '<h', 'default': 0}),
+        ('prts',
+            {'dt': 'multi', 'nField': 'nPrt', 'default': [
+                OrderedDict([
+                    ('filename', {'dt': '<s', 'default': '', 'value': ''})
+                ])
+            ]}
+         ),
+        ('currentPrt', {'dt': '<h', 'default': 0}),
+        ('datatype', {'dt': '<h', 'default': 2}),
+        ('volumes', {'dt': '<h', 'default': 0}),
+        ('Resolution', {'dt': '<h', 'default': 3}),
+        ('XStart', {'dt': '<h', 'default': 57}),
```

Comment:
Yes, there (kind of) is: http://support.brainvoyager.com/installation-introduction/23-file-formats/379-users-guide-23-the-format-of-vtc-files.html

Look at the 'default' column. Also, NeuroElf implements it this way.